repl/src/main/scala/com/cloudera/livy/repl/Session.scala

```diff
@@ -51,7 +54,9 @@ class Session(interpreter: Interpreter)
   private implicit val formats = DefaultFormats
 
   private var _state: SessionState = SessionState.NotStarted()
-  private var _history = IndexedSeq[Statement]()
+  private val _statements = mutable.Map[Int, Statement]()
```

Comment:
Repl was using Guava and caused a lot of jar collision problems with user jars. We don't want to shade Guava either because it's pretty large.
Mind if we keep this PR solely for recovery and deal with this later?